src/index.ts

```diff
@@ -1,107 +1,106 @@
+
+      // Infer config from dataset
+      const dataset = el.dataset;
+      if (dataset.unfeedOpen !== undefined) open(el);
+      if (dataset.unfeedFooter !== undefined)
+        config.footer = dataset.unfeedFooter;
+
+      // Customize primary color
+      if (dataset.unfeedPrimaryColor) {
+        (document.querySelector(":root") as HTMLElement).style.setProperty(
+          "--unfeed-primary-color",
+          dataset.unfeedPrimaryColor
+        );
+      }
+
+      if (dataset.unfeedButton) {
+        el.addEventListener("click", (e: Event) =>
+          open(e.target as HTMLElement)
+        );
+      } else {
+        buildHtml(containerElement, el);
+      }
+    });
 }
 window.addEventListener("load", init);
 
-const containerElement = document.createElement("div");
-containerElement.id = "unfeed__container";
-
 const trap = createFocusTrap(containerElement, {
-  initialFocus: "#unfeed__radio--issue",
+  initialFocus: `#${ns}radio--0`,
   allowOutsideClick: true,
 });
 
-function renderHtml(containerElement: HTMLDivElement) {
-  document.body.appendChild(containerElement);
-  containerElement.innerHTML = formHTML(locale);
+function buildHtml(
+  containerElement: HTMLDivElement,
+  parent: HTMLElement,
+  getDataset: () => DOMStringMap = () => parent.dataset
+) {
+  parent.appendChild(containerElement);
+  containerElement.innerHTML = formHTML(ns, config.locale);
   containerElement.style.display = "block";
+  containerElement
+    .querySelector(`#${ns}close`)!
+    .addEventListener("click", close);
+
+  const mergeConfig = { ...config, ...(window as any).unfeed } as typeof config;
+
+  // Build options
+  const optionsElement = containerElement.querySelector(
+    `#${ns}radio-group`
+  ) as HTMLElement;
+  optionsElement.innerHTML = mergeConfig.options
+    .map(({ icon, label, name, placeholder }, i) => [
+      `<input
+      class="${ns}radio"
+      type="radio"
+      id="${ns}radio--${i}"
+      name="feedbackType"
+      value="${name || label}"
+      data-placeholder="${placeholder || ""}"
+      required />
+    <label for="${ns}radio--${i}" class="${ns}button ${ns}radio-label">
+      ${icon ? `<span class="${ns}radio-icon">${icon}</span>` : ""}${label}
+    </label>`,
+    ])
+    .join("");
+
+  Array.from(optionsElement.getElementsByClassName(`${ns}radio`)).forEach(
+    (el) => el.addEventListener("change", changeType)
+  );
+
+  (containerElement as Element)
+    .querySelector(`#${ns}form`)!
+    .addEventListener("submit", (e) => submit(e, getDataset));
 
   // Customize footer
-  if (config.footer !== undefined) {
-    containerElement.querySelector("#unfeed__footer")!.innerHTML =
-      config.footer;
+  if (mergeConfig.footer !== undefined) {
+    containerElement.querySelector(`#${ns}footer`)!.innerHTML =
+      mergeConfig.footer;
   }
 }
 
 function open(target: HTMLElement) {
-  renderHtml(containerElement);
+  buildHtml(containerElement, document.body, () => target.dataset);
```

Comment:
By default, widget will use parent.dataset to get latest dataset before submission. For widget that triggered with a button, we'll use document.body as parent and thus we need to use custom dataset getter (`() => target.dataset`).